### PR TITLE
GPC-145: Change event response to always include allowed fields

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt [hunk 1]
```diff
@@ -52,4 +57,72 @@ data class DeathNotificationDetails(
...
+        )
+      }
+    }
+  }
+}
+
+class DeathNotificationDetailsSerializer :
+  StdSerializer<DeathNotificationDetails>(DeathNotificationDetails::class.java) {
+  override fun serialize(value: DeathNotificationDetails?, gen: JsonGenerator?, provider: SerializerProvider?) {
+    gen?.writeStartObject()
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt:87]: not having a generator here feels like a problem, can we check it for null once and then just use it throughout?

File: src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt [hunk 2]
```diff
@@ -52,4 +57,72 @@ data class DeathNotificationDetails(
...
+    value?.run {
+      if (enrichmentFields.contains("registrationDate")) {
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt:89]: I can't help thinking there's got to be a way to make this less branchy logic, but I'm not sure at all what that is :)

File: src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt [hunk 3]
```diff
@@ -52,4 +57,72 @@ data class DeathNotificationDetails(
...
+        gen?.writeNullableStringField("firstNames", firstNames)
+      }
+      if (enrichmentFields.contains("lastName")) {
+        gen?.writeNullableStringField("lastName", lastName)
+      }
+      if (enrichmentFields.contains("sex")) {
+        gen?.writeNullableStringField("sex", sex?.jsonName)
+      }
+      if (enrichmentFields.contains("dateOfDeath")) {
+        gen?.writeNullableStringField("dateOfDeath", dateOfDeath?.toString())
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt:102]: Do these serialize as we'd expect or should we format the date ourselves?

Comment (olevett) [reply src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt:102]: For example would it be better to do this date->string/json mapping within our writeNullableStringField bit instead?

**Review by olevett — APPROVED** [review]

File: src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt (hunk 3 above)
Comment (ethanmills) [reply src/main/kotlin/uk/gov/gdx/datashare/models/DeathNotificationDetails.kt:102]: Yeah this comes out fine. It gets messy when you start adding times, but date works

**Pull request merged**